endpoints/s3/src/putobject.cpp

```diff
@@ -58,64 +98,135 @@ void manually_parse_chunked_body_write_to_irods_in_background(
 	parsing_state current_state,
 	size_t chunk_size,
 	const std::string& parsing_buffer_string,
+	bool know_part_offset,
+	bool keep_dstream_open_flag,
 	const std::string func);
 
-/*std::make_shared<incremental_async_read>(
-    parser, session_ptr, response, path, upload_part, upload_part_filename)
-        ->start();*/
 class incremental_async_read : public std::enable_shared_from_this<incremental_async_read>
 {
 	irods::http::session_pointer_type session_ptr_;
 	beast::http::response<beast::http::empty_body> resp_;
 	std::shared_ptr<beast::http::request_parser<boost::beast::http::buffer_body>> parser_;
 	std::string irods_path_;
 	bool upload_part_flag_;
+	bool know_part_offset_flag_;
```

Comment:
Does this mean that we (a) *want to know* the part offset, we (b) *do know* the part offset, or that we (c) *want to know whether we know* the part offset? It looks like it's (b). Should this be called `part_offset_is_known_`?

Reply:
It is b and will change to that.

Reply:
done